packages/validator/src/signer/local.ts

```diff
@@ -0,0 +1,57 @@
+import {PointFormat, SecretKey} from "@chainsafe/bls";
+import {BLSPubkey} from "@chainsafe/lodestar-types";
+import {toHexString} from "@chainsafe/ssz";
+import {PubkeyHex} from "../types";
+
+export interface ISigner {
+  sign(pubkey: BLSPubkey | PubkeyHex, signingRoot: Uint8Array): Uint8Array;
+  addKey(secretKey: SecretKey): void;
+  removeKey(pubkey: BLSPubkey | PubkeyHex): boolean;
+  hasKey(pubkeyHex: PubkeyHex): boolean;
+  hasSomeKeys(): boolean;
+  getKeys(): PubkeyHex[];
+}
+
+/**
+ * Signs messages while hidding secret keys from consumers.
+ * Does not use ES6 classes to implement really private properties, enforced at runtime.
```

Comment:
Maybe say: "Uses a closure to implement ..."

Reply:
Updated as suggested